ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StateRef.kt

```diff
@@ -0,0 +1,18 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+
+/**
+ * Defines a reference to a [ContractState].
+ *
+ * @property index The index of the state in the transaction's outputs in which the referenced state was created.
+ * @property transactionHash The hash of the transaction in which the referenced state was created.
+ */
+@DoNotImplement
+@CordaSerializable
+interface StateRef {
```

Comment:
I feel like we need to nail down what can be a POJO and what can't, as I've heard different opinions. `StateRef`, `StateAndRef`, `TransactionState`, `TimeWindow`, etc were all POJOs in a previous PR, but were changed to interfaces with implementations in Corda Runtime as per a conversation I had with Christian (which went something along the lines of "If user code isn't expected to instantiate these things, then they can be interfaces, and let the runtime deal with it).

Yes we can reorder/rename, although I don't think the order matters so much on the interface as it does on the implementation. The reason it was changed from `txhash` to `transactionHash` was basically because I'm a stickler for naming, and "don't abbreviate" is high on my list of don't dos.

Reply:
I agree on the non abbreviated naming.
With modern IDEs that has become a non-problem.
Also, if you have a proper and non-abbreviated names - you can sometimes save yourself the comments (if the name is self-explanatory).

Reply:
I stand by that anything users don't need to instantiate themselves should be an interface - `StateRef` might not be one of those things, though.